File: 189-rotate-array/rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        k = k%len(nums)
        l = 0
        r = len(nums)-1
        while l<r:
            tmp = nums[r]
            nums[r] = nums[l]
            nums[l] = tmp
            l+=1
            r-=1
        l = 0
        r = k-1
        while l<r:
            tmp = nums[r]
            nums[r] = nums[l]
            nums[l] = tmp
            l+=1
            r-=1
        l = k
        r = len(nums)-1
        while l<r:
            tmp = nums[r]
            nums[r] = nums[l]
            nums[l] = tmp
            l+=1
            r-=1
            

File: 189-rotate-array/rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let n = nums.length
    k%=n

    const reverse =(l,r) =>{
while(l < r){
     [nums[l] , nums[r]] = [nums[r] , nums[l]];
 l++
 r--
}
    }
    reverse(0,n-1);
    reverse(0,k-1);
    reverse(k,n-1)
};